taipy/core/_orchestrator/_dispatcher/_development_job_dispatcher.py

```diff
@@ -44,5 +45,7 @@ def _dispatch(self, job: Job):
         Parameters:
             job (Job^): The job to submit on an executor with an available worker.
         """
+        job.execution_started_at = datetime.datetime.now()
```

Comment:
What do you think about creating a context manager to set execution_started_at and execution_end_at? Something like:

```
import datetime
from contextlib import contextmanager

@contextmanager
def track_execution(job):
    job.execution_started_at = datetime.datetime.now()
    try:
        yield
    finally:
        job.execution_ended_at = datetime.datetime.now()
   ```
 And them:
 ```
   def _dispatch(self, job: Job):
    """Dispatches the given `Job` on an available worker for execution.

    Parameters:
        job (Job): The job to submit on an executor with an available worker.
    """
    with track_execution(job):
        rs = _TaskFunctionWrapper(job.id, job.task).execute()
        self._update_job_status(job, rs)